lib/ClimaLandSimulations/src/utilities/domain_setup.jl

```diff
@@ -1,14 +1,55 @@
-nelements = 20
-zmin = FT(-10)
-zmax = FT(0)
-h_canopy = h_stem + h_leaf
-compartment_midpoints =
-    n_stem > 0 ? [h_stem / 2, h_stem + h_leaf / 2] : [h_leaf / 2]
-compartment_surfaces = n_stem > 0 ? [zmax, h_stem, h_canopy] : [zmax, h_leaf]
-
-land_domain = Column(;
-    zlim = (zmin, zmax),
-    nelements = nelements,
-    dz_tuple = (dz_bottom, dz_top),
+export domain_setup
+
+"""
+    domain_setup(site_setup_out;
+        nelements = FT(20),
+        zmin = FT(-10),
+        zmax = FT(0)
+        )
+
+setup the soil and canopy domain of the simulation. Returns:
+- h_canopy,
+- compartment_midpoints, 
+- compartment_surfaces,
+- land_domain,
+- canopy_domain,
```

Comment:
Could you briefly explain these? e.g. what is a component

Reply:
hmmm also we are redefining things that we already have defined and documented in ClimaLand/src.

Reply:
I think it is okay to double the docstrings definition, if there's different userbase for ClimaLand and ClimaLandSimulation (which just calls ClimaLand), but in the docs it should be defined just once, in ClimaLand. 
Alternatively, we could just link to ClimaLand docs in the docstrings. 

I will leave this for another PR, we can decide later